built-in-services/multi-signature/src/lib.rs

```diff
@@ -0,0 +1,554 @@
+                owner: payload.owner,
+                threshold: payload.threshold,
+                memo: payload.memo,
+            };
+
+            self.sdk.set_account_value(&address, 0u8, permission);
+            ServiceResponse::<GenerateMultiSigAccountResponse>::from_succeed(
+                GenerateMultiSigAccountResponse { address },
+            )
+        } else {
+            ServiceResponse::<GenerateMultiSigAccountResponse>::from_error(
+                112,
+                "generate address from tx_hash failed".to_owned(),
+            )
+        }
+    }
+
+    #[cycles(100_00)]
+    #[read]
+    fn get_account_from_address(
+        &self,
+        _ctx: ServiceContext,
+        payload: GetMultiSigAccountPayload,
+    ) -> ServiceResponse<GetMultiSigAccountResponse> {
+        if let Some(permission) = self.sdk.get_account_value(&payload.multi_sig_address, &0u8) {
+            ServiceResponse::<GetMultiSigAccountResponse>::from_succeed(
+                GetMultiSigAccountResponse { permission },
+            )
+        } else {
+            ServiceResponse::<GetMultiSigAccountResponse>::from_error(
+                113,
+                "account not existed".to_owned(),
+            )
+        }
+    }
+
+    #[cycles(210_00)]
+    #[read]
+    fn verify_signature(
+        &self,
+        ctx: ServiceContext,
+        payload: VerifySignaturePayload,
+    ) -> ServiceResponse<()> {
+        let pubkeys = payload.pubkeys.clone();
+        let signatures = payload.signatures.clone();
+
+        if pubkeys.len() != signatures.len() {
+            return ServiceResponse::<()>::from_error(
+                114,
+                "pubkkeys len is not equal to signatures len".to_owned(),
+            );
+        }
+
+        if pubkeys.len() > MAX_PERMISSION_ACCOUNTS as usize {
+            return ServiceResponse::<()>::from_error(
+                115,
+                "len of signatures must be [1,16]".to_owned(),
+            );
+        }
+
+        if pubkeys.len() == 1 {
+            if let Ok(addr) = Address::from_pubkey_bytes(pubkeys[0].clone()) {
+                if addr == payload.sender {
+                    return self._verify_single_signature(
+                        &ctx.get_tx_hash().unwrap(),
+                        &signatures[0],
+                        &pubkeys[0],
+                    );
+                }
+            } else {
+                return ServiceResponse::<()>::from_error(123, "invalid public key".to_owned());
+            }
+        }
+
+        let mut recursion_depth = 0u8;
+        self._verify_multi_signature(
+            &ctx.get_tx_hash().unwrap(),
+            &Witness::new(pubkeys, signatures).to_addr_map(),
+            &payload.sender,
+            &mut recursion_depth,
+        )
+    }
+
+    fn _verify_multi_signature(
+        &self,
+        tx_hash: &Hash,
+        wit_map: &HashMap<Address, (Bytes, Bytes)>,
+        sender: &Address,
+        recursion_depth: &mut u8,
```

Comment:
It should not be &mut u8 here.
In this case, recursion_depth recorded the num of all multiSig accounts in the deep search tree, not the depth.